### PR TITLE
[IMP] udes_sale_stock: Restrict Outbound and Stock user access levels

### PR DESCRIPTION
**Pull request by richardwhitefoot**
Signed-off-by: rjw <richard.whitefoot@unipart.io>

### CODE REVIEW

File: addons/udes_sale_stock/models/partner_pii.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        result = super(ResPartner, self).name_get()
+
+        if self.env.uid == SUPERUSER_ID or self.env.user.u_view_customers:
+            return result
+
+        for i, record in enumerate(result):
+            if isinstance(record, tuple) and \
+                self.sudo().search_count([('id', '=', record[0]),('customer', '=', True)]):
+                result[i] = (record[0], REDACTED)
```
Comment (parisiale) [inline addons/udes_sale_stock/models/partner_pii.py]: Not a big fan of modifying a collection while iterating it. Could you create a new one to be used as the return value? You should be using references anyway, so it should be ok.

File: addons/udes_sale_stock/models/partner_pii.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+def _read(self, result):
+    """Redact fields based on res.user.u_view_customers"""
+
+    if self.env.uid == SUPERUSER_ID or self.env.user.u_view_customers:
+        return result
+
+    for record in result:          
+        for field in record:
+            if field in PII_RELATIONS and isinstance(record[field], tuple) and \
+                self.env['res.partner'].sudo().search_count([('id', '=', record[field][0]),('customer', '=', True)]):
```
Comment (parisiale) [inline addons/udes_sale_stock/models/partner_pii.py]: Too long. Besides, this bit of the if clause is aligned with the actual block which makes it confusing.

Comment (parisiale) [reply addons/udes_sale_stock/models/partner_pii.py]: Also, need space between tuples.

File: addons/udes_sale_stock/models/partner_pii.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                record[field] = (record[field][0], REDACTED)
+                _logger.info("Redacted: %s.%s", self._name, field)
```
Comment (parisiale) [inline addons/udes_sale_stock/models/partner_pii.py]: Do we need the l10n helper (the `_` underscore function)?

Comment (richardwhitefoot) [reply addons/udes_sale_stock/models/partner_pii.py]: Added in rather than constant

File: addons/udes_sale_stock/models/partner_pii.py (hunk 1 above)
Comment (richardwhitefoot) [reply addons/udes_sale_stock/models/partner_pii.py]: OC, will change it

**Pull request merged**